allennlp/service/predictors/wikitables_parser.py

```diff
@@ -36,3 +57,42 @@ def _json_to_instance(self, json_dict: JsonDict) -> Tuple[Instance, JsonDict]:
                                                          tokenized_question=tokenized_question)
         extra_info = {'question_tokens': tokenized_question}
         return instance, extra_info
+
+    @overrides
+    def predict_json(self, inputs: JsonDict, cuda_device: int = -1) -> JsonDict:
+        instance, return_dict = self._json_to_instance(inputs)
+        outputs = self._model.forward_on_instance(instance, cuda_device)
+
+        logical_form_filename = os.path.join(SEMPRE_DIR, 'logical_forms.txt')
```

Comment:
I'd recommend pulling out this answer-getting logic into its own method, with a clear name and signature (e.g., pass in `outputs['logical_form']`, `inputs['table']`, etc.).  It'd make this code easier to read and understand.  You might give a note there, also, that the jar that we have isn't optimal for this use case - we're using a script designed for computing accuracy, and just pulling out a piece of it.  Writing a new entry point to the jar that's tailored for this use would be cleaner.  Feel free to put that in a `TODO(matt):` block.

Reply:
done.